core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -488,7 +488,10 @@ public boolean doAuthenticate(String userName, Object credential) throws UserSto
 
         String userAttributeSeparator = ",";
         String userDN = null;
-        LdapName ldn = (LdapName)userCache.get(userName);
+        LdapName ldn = null;
+        if (userName != null) {
```

Comment:
It may be good to validate username once at the begining and then perform further checking instead of validating it again and again, WDYT ?